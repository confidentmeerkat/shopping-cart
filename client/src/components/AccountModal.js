import React from 'react';
import Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import '../styles/AccountModal.css';

class AccountModal extends React.Component {

  state = {
    email: this.props.user.email,
    address: this.props.user.address,
    phone: this.props.user.phone
  }

  onInputChange = (e) => {
    const value = e.target.value;
    const key = e.target.name;

    this.setState({ [key]: value });
  }

  render() {
    return (
      <Modal
        className="account-modal"
        isOpen={this.props.isOpen}
        onRequestClose={this.props.onRequestClose}
      >
        <form className="form" action="/api/user" method="POST">
          <h1>Edit Account</h1>
          <TextField
            hintText="Enter E-mail"
            floatingLabelText="E-mail"
            name="email"
            value={this.state.email}
            onChange={this.onInputChange}
          /><br />
          <TextField
            hintText="Enter Address"
            floatingLabelText="Address"
            name="address"
            value={this.state.address}
            onChange={this.onInputChange}
          /><br />
          <TextField
            hintText="Enter Telephone Number"
            floatingLabelText="Telephone Number"
            name="phone"
            value={this.state.phone}
            onChange={this.onInputChange}
          /><br />
          <RaisedButton
            className="btn"
            label="Save"
            primary={true}
            type="submit"
          />
        </form>
      </Modal>
    );
  }
}

export default AccountModal;