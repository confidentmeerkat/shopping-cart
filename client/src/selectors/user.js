export const selectUser = (state) => state.loggedUser.user;