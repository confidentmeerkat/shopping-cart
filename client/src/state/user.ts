import { Order } from './order';

export interface User {
  orders: Order[];
  username: string;
  email: string;
  address: string;
  phone: string;
  _id: string;
}
