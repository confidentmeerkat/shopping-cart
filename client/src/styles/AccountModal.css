.account-modal {
  width: 30%;
  height: 500px;
  margin: auto;
  margin-top: 100px;
  background: white;
  border: 1px solid #00BCD4;
  border-radius: 5px;
  outline: none;
}

.account-modal .form {
  width: 70%;
  margin: auto;
  margin-top: 10px;
  text-align: center;
  font-family: 'Roboto';
}

.account-modal .form h1 {
  color: #00BCD4;
}

.account-modal .form .btn {
  margin: 30px auto 50px auto;
}