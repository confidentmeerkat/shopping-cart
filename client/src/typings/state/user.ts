import { IOrder } from './order';

export interface IUser {
  orders: IOrder[];
  username: string;
  email: string;
  address: string;
  phone: string;
  _id: string;
}
