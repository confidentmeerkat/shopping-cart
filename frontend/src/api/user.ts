import http from './index';

const PREFIX = '/user';

export const getUser = () => http.get(PREFIX);

export const editUser = (data: Record<string, any>) => http.put(PREFIX, data);
