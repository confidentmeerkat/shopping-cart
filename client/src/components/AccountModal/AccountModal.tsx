import * as React from 'react';
import * as Modal from 'react-modal';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import { User } from '@state/index';
import '@styles/AccountModal.css';

interface Props {
  user: User;
  isOpen: boolean;
  onRequestClose: () => void;
}

interface State {
  email: string;
  address: string;
  phone: string;
}

class AccountModal extends React.Component<Props, State> {
  state = {
    email: this.props.user.email,
    address: this.props.user.address,
    phone: this.props.user.phone
  }

  onInputChange = (e: React.FormEvent<HTMLInputElement>) => {
    const value = e.currentTarget.value;
    const key = e.currentTarget.name;

    this.setState((prevState: State) => ({
      ...prevState,
      [key]: value
    }));
  }

  render() {
    return (
      <Modal
        className="account-modal"
        isOpen={this.props.isOpen}
        onRequestClose={this.props.onRequestClose}
      >
        <form className="form" action="/api/user" method="POST">
          <h1>Edit Account</h1>
          <TextField
            id="email"
            hintText="Enter E-mail"
            floatingLabelText="E-mail"
            name="email"
            value={this.state.email}
            onChange={this.onInputChange}
          /><br />
          <TextField
            id="address"
            hintText="Enter Address"
            floatingLabelText="Address"
            name="address"
            value={this.state.address}
            onChange={this.onInputChange}
          /><br />
          <TextField
            id="phone"
            hintText="Enter Telephone Number"
            floatingLabelText="Telephone Number"
            name="phone"
            value={this.state.phone}
            onChange={this.onInputChange}
          /><br />
          <RaisedButton
            className="btn"
            label="Save"
            primary={true}
            type="submit"
          />
        </form>
      </Modal>
    );
  }
}

export default AccountModal;
