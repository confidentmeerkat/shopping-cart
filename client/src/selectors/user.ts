import { IState, IUser } from '@typings/state/index';

export const selectUser = (state: IState): IUser | null => state.loggedUser.user;
