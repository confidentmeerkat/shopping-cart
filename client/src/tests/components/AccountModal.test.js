import React from 'react';
import { shallow } from 'enzyme';
import AccountModal from '../../components/AccountModal';
import user from '../__mocks__/user';

describe('<AccountModal />', () => {
  it('renders AccountModal correctly', () => {
    const wrapper = shallow(<AccountModal user={user} isOpen={true} onRequesClose={() => { }} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('sets state on input change correctly', () => {
    const wrapper = shallow(<AccountModal user={user} isOpen={true} onRequesClose={() => { }} />);

    wrapper.find('#email').simulate('change', { target: { name: 'email', value: 'xxx@mail.com' } });
    wrapper.find('#address').simulate('change', { target: { name: 'addres', value: '1st street' } });
    wrapper.find('#phone').simulate('change', { target: { name: 'phone', value: '123456' } });

    expect(wrapper.state('email')).toEqual('xxx@mail.com');
    expect(wrapper.state('address')).toEqual('1st street');
    expect(wrapper.state('phone')).toEqual('123456');
  });
});