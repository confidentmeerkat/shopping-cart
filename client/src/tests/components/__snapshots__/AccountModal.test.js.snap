// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountModal /> renders AccountModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="account-modal"
  closeTimeoutMS={0}
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <form
    action="/api/user"
    className="form"
    method="POST"
  >
    <h1>
      Edit Account
    </h1>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="E-mail"
      fullWidth={false}
      hintText="Enter E-mail"
      id="email"
      multiLine={false}
      name="email"
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="john@mail.com"
    />
    <br />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Address"
      fullWidth={false}
      hintText="Enter Address"
      id="address"
      multiLine={false}
      name="address"
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="1st street"
    />
    <br />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Telephone Number"
      fullWidth={false}
      hintText="Enter Telephone Number"
      id="phone"
      multiLine={false}
      name="phone"
      onChange={[Function]}
      rows={1}
      type="text"
      underlineShow={true}
      value="123 456"
    />
    <br />
    <RaisedButton
      className="btn"
      disabled={false}
      fullWidth={false}
      label="Save"
      labelPosition="after"
      primary={true}
      secondary={false}
      type="submit"
    />
  </form>
</Modal>
`;
